backend/src/vscode-youi-events.ts

```diff
@@ -4,17 +4,31 @@ import { YouiEvents } from "./youi-events";
 import { IRpc } from "@sap-devx/webview-rpc/out.ext/rpc-common";
 import { GeneratorFilter, GeneratorType } from './filter';
 import { relative, isAbsolute } from 'path';
+import { GeneratorOutput } from './vscode-output';
+import { IChildLogger } from '@vscode-logging/logger';
+import { getClassLogger } from './logger/logger-wrapper';
 
 export class VSCodeYouiEvents implements YouiEvents {
     private webviewPanel: vscode.WebviewPanel;
     private readonly genFilter: GeneratorFilter;
     private readonly messages: any;
     private resolveFunc: any;
+    private output: GeneratorOutput;
+    private isOutputShown: boolean;
+    private readonly logger: IChildLogger;
 
-    constructor(rpc: IRpc, webviewPanel: vscode.WebviewPanel, genFilter: GeneratorFilter, messages: any) {
+    constructor(
+        rpc: IRpc, 
+        webviewPanel: vscode.WebviewPanel, 
+        genFilter: GeneratorFilter, 
+        messages: any, 
+        output: GeneratorOutput) {
         this.webviewPanel = webviewPanel;   
         this.genFilter = genFilter;
         this.messages = messages;    
+        this.output = output;
+        this.isOutputShown = false;
```

Comment:
This logic will not work (isOutputShown )
Output could be opened from UI in any stage

Reply:
I know what you mean, but this is Avital's requirement. Verifying with her.

Reply:
Avital approved that the current logic is by her request and she wants it as now: in case the generator calls showProgress() several times, show the progress notification each time, only in case the user clicked "Progress" stop show the next notifications.